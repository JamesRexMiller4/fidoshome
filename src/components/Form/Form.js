import React, { Component } from 'react';
import './Form.scss';
import { addAnimals } from '../../actions';
import { connect } from 'react-redux';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      type: 'dog',
      state: '',
      zipcode: '',
    }
  }
  
  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value.toLowerCase()})
  }

  onSubmit = () => {
    let type = this.state.type;
    let state = this.state.state;
    let zip = this.state.zipcode;
    let location = 'location=';

    state.length > zip.length ? location = location + state
    : zip.length > state.length ? location = location + zip
    : location = ''

    fetch('https://api.petfinder.com/v2/animals?' + location + '&status=adoptable' + '&type=' + type, {
      headers: {
        'Authorization': 'Bearer ' + sessionStorage.getItem('token'),
        'Content-Type': 'application/json',
        'Origin': 'localhost:3000/home'
      }
    })
    .then(res => res.json())
    .then(data => {this.props.addAnimals(data.animals)})
  }

  render() { 
    return ( 
      <form className='form'>
        <h2 className='form-h2'>Search</h2>
        <label for='type-input' className='label'>Type</label>
        <select onChange={(e) => this.handleChange(e)} id='type-input' className='form-input' type='text' name='type' placeHolder='Dog'>
          <option name='type'>Dog</option>
          <option name='type'>Cat</option>
          <option name='type'>Rabbit</option>
        </select>
        <label for='state-input' className='label'>State</label>
        <input onChange={(e) => this.handleChange(e)} id='state-input' className='form-input' type='text' name='state' placeHolder='CO'></input>
        <label for='zipcode-input' className='label'>Zipcode</label>
        <input onChange={(e) => this.handleChange(e)} id='zipcode-input' className='form-input' type='text' name='zipcode' placeHolder='80042'></input>
        <button onClick={this.onSubmit}type='button' className='form-btn'>Find Pets</button>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addAnimals: animals => dispatch(addAnimals(animals))
})

export default connect(null, mapDispatchToProps)(Form);