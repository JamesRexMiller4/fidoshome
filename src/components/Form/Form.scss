form {
  width: 30vw;
  height: 80vh;
  margin: 30px;
  padding: 30px;
  display: flex;
  flex-direction: column;
  background-color: #710627;
  border: 6px solid;
  border-radius: 3px;
  border-color: #a20839;
}

.form-h2 {
  color: white;
  font-family: 'Libre Baskerville', serif;
  font-weight: bold;
  font-size: 40px;
  margin-bottom: 30px;
}

.label {
  color: white;
  font-size: 20px;
  font-family: 'Libre Baskerville', serif;
  text-align: justify;
  margin-bottom: 5px;
  margin-left: 15px;
}

.form-input {
  width: 90%;
  height: 6%;
  background-color: white;
  font-family: 'Libre Baskerville', serif;
  font-size: 15px;
  margin-left: 15px;
  margin-bottom: 20px;
    &:focus {
      outline: none;
      border: 2px solid;
      border-color: rgb(205, 174, 2);
    }
}

.form-btn {
  width: 90%;
  height: 8%;
  margin-top: 90px;
  margin-left: 15px;
  color: white;
  font-family: 'Libre Baskerville', serif;
  font-weight: bold;
  font-size: 30px;
  background-color: rgba(69, 15, 179, 0.844);
  border: 3px solid;
  border-color: rgba(96, 39, 210, 0.844);

  &:hover {
    color: rgb(242, 212, 46);
    cursor: pointer;
    border: 4px solid;
  }
}