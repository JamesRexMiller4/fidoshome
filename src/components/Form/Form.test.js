import React from 'react';
import {shallow} from 'enzyme';
import { Form, mapDispatchToProps } from './Form';
import VeryGoodBoy from '../../images/verygoodboy.PNG';
import { addAnimals } from '../../actions/index';

describe('Form', () => {
let wrapper, mockData;

  beforeEach(() => {
    wrapper = shallow(<Form />)
    mockData = [{
      photos: [{medium: VeryGoodBoy}],
      id: 4, 
      name: 'Fido',
      age: 'Older than time itself',
      contact: {address: {city: 'Venice'}}
    }];
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default local state', () => {
    const mockState = {
      type: 'dog',
      state: '',
      zipcode: '',
      page: 1
    };
    expect(wrapper.instance().state).toEqual(mockState);
  });
  describe('handleChange', () => {
    it('should update state when handleChange is called', () => {
      const mockEvent = {target: {name: 'state', value: 'CO'}}
      const mockState = {
        type: 'dog',
        state: '',
        zipcode: '',
        page: 1
      };
      const expectedState = {
        type: 'dog',
        state: 'co',
        zipcode: '',
        page: 1
      };
      expect(wrapper.instance().state).toEqual(mockState);
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.instance().state).toEqual(expectedState);
    });
  });
  describe('onSearch', () => {
  it('should call fetchAnimals when onSearch is called from a user click', async () => {
      const apiCalls = require('../../apiCalls/apiCalls');
      const fetchSpy = jest.spyOn(apiCalls, 'fetchAnimals');
      fetchSpy.mockResolvedValue();

      wrapper.find('.form-btn').simulate('click');

      await Promise.resolve();
      
      expect(fetchSpy).toHaveBeenCalled();
    });
  });
  describe('mapDispatchToProps', () => {
    it('should call addAnimals with dispatch', () => {
      const mockDispatch = jest.fn();
      const mockActionToDispatch = addAnimals(mockData);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addAnimals(mockData);
      expect(mockDispatch).toHaveBeenCalledWith(mockActionToDispatch);
    });
  });
});