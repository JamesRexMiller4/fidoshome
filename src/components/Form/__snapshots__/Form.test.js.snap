// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="form"
>
  <h2
    className="form-h2"
  >
    Search
  </h2>
  <label
    className="label"
    for="type-input"
  >
    Type
  </label>
  <select
    className="form-input"
    id="type-input"
    name="type"
    onChange={[Function]}
    placeHolder="Dog"
    type="text"
  >
    <option
      name="type"
    >
      Dog
    </option>
    <option
      name="type"
    >
      Cat
    </option>
    <option
      name="type"
    >
      Rabbit
    </option>
  </select>
  <label
    className="label"
    for="state-input"
  >
    State
  </label>
  <input
    className="form-input"
    id="state-input"
    name="state"
    onChange={[Function]}
    placeHolder="CO"
    type="text"
  />
  <label
    className="label"
    for="zipcode-input"
  >
    Zipcode
  </label>
  <input
    className="form-input"
    id="zipcode-input"
    name="zipcode"
    onChange={[Function]}
    placeHolder="80042"
    type="text"
  />
  <button
    className="form-btn"
    onClick={[Function]}
    type="button"
  >
    Find Pets
  </button>
</form>
`;
